spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -440,8 +440,25 @@ trait OptimisticTransactionImpl extends TransactionalWrite
   // then it will be detected as a conflict and the transaction will anyway fail.
   private[delta] val readSnapshotTableCommitCoordinatorClientOpt:
     Option[TableCommitCoordinatorClient] = {
-      snapshot.tableCommitCoordinatorClientOpt
-  }
+      val coordinatorOpt = snapshot.tableCommitCoordinatorClientOpt
+      val coordinatorName =
+        DeltaConfigs.COORDINATED_COMMITS_COORDINATOR_NAME.fromMetaData(snapshot.metadata)
+      if (snapshot.version != -1 && coordinatorName.isDefined && coordinatorOpt.isEmpty) {
+        recordDeltaEvent(
+          deltaLog,
+          CoordinatedCommitsUsageLogs.COMMIT_COORDINATOR_MISSING_IMPLEMENTATION_WRITE,
+          data = Map(
+            "commitCoordinatorName" -> coordinatorName.get,
+            "readVersion" -> readVersion.toString
+          )
+        )
+        throw new IllegalStateException(
```

Comment:
User could face this error. This should follow the error class framework. We should fix this in same PR or a followup PR.

Reply:
Done